tests/unit/nupic/serializable_test.py

```diff
@@ -0,0 +1,80 @@
+# ----------------------------------------------------------------------
+# Numenta Platform for Intelligent Computing (NuPIC)
+# Copyright (C) 2017, Numenta, Inc.  Unless you have an agreement
+# with Numenta, Inc., for a separate license for this software code, the
+# following terms and conditions apply:
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero Public License version 3 as
+# published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+# See the GNU Affero Public License for more details.
+#
+# You should have received a copy of the GNU Affero Public License
+# along with this program.  If not, see http://www.gnu.org/licenses.
+#
+# http://numenta.org/licenses/
+# ----------------------------------------------------------------------
+
+import os
+import unittest
+
+import capnp
+import serializable_test_capnp
+
+from nupic.serializable import  Serializable
+
+
+
+class SerializableTest(unittest.TestCase):
+
+  def testABCProtocolEnforced(self):
+
+    class Foo(Serializable):
+      pass # read(), write(), getCapnpSchema() not implemented here
+
+    with self.assertRaises(TypeError):
+      Foo()
+
+
+  def testReadFromAndWriteToFile(self):
+    """ Test generic usage of serializable mixin class """
+
+    class Bar(object):
+      pass
+
+
+    class Foo(Bar, Serializable):
```

Comment:
You can (see testABCProtocolEnforced).  I added it in to show (and provide imolicit test coverage) that it can be used as a mixin.

I will have to verify, but iirc `class Foo(object, Serializable):` fails.